opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```diff
@@ -148,10 +139,10 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         if (servletRequest.getAttribute(SERVER_SPAN_CONTEXT) != null) {
             chain.doFilter(servletRequest, servletResponse);
         } else if (isTraced(httpRequest, httpResponse)){
-            SpanContext extractedContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
+            SpanContext extractedContext = GlobalTracer.get().extract(Format.Builtin.HTTP_HEADERS,
                     new HttpServletRequestExtractAdapter(httpRequest));
 
-            final Span span = tracer.buildSpan(httpRequest.getMethod())
+            final Span span = GlobalTracer.get().buildSpan(httpRequest.getMethod())
```

Comment:
if the goal is to allow GlobalTracer to return different tracers at different times, it's still doable forcing the hard dependency, just capture the fact in a lambda

```java
interface TracerProvider {
  Tracer getTracer()
}

public TracingFilter() {
   this(() => { 
     GlobalTracer.get(); 
   });
}

public TracingFilter(final Tracer tracer) {
  this(() = > {
     return tracer;
  });
}

private TracingFilter(TracerProvider tracerProvider) {
   // actual constructor
}
```